res/controllers/Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -0,0 +1,782 @@
+MixtrackProFX.waveformsSynced = true;
+
+// jogwheel
+MixtrackProFX.jogScratchSensitivity = 1024;
+MixtrackProFX.jogScratchAlpha = 1; // do NOT set to 2 or higher
+MixtrackProFX.jogScratchBeta = 1/32;
+MixtrackProFX.jogPitchSensitivity = 10;
+MixtrackProFX.jogSeekSensitivity = 10000;
+
+// blink settings
+MixtrackProFX.enableBlink = true;
+MixtrackProFX.blinkDelay = 700;
+
+// autoloop sizes, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatloop_X_toggle
+MixtrackProFX.autoLoopSizes = [
+    "0.0625",
+    "0.125",
+    "0.25",
+    "0.5",
+    "1",
+    "2",
+    "4",
+    "8"
+];
+
+// beatjump values, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatjump_X_forward
+// underscores (_) at the end are needed because numeric values (e.g. 8) have two underscores (e.g. beatjump_8_forward),
+// but "beatjump_forward"/"beatjump_backward" have only one underscore
+MixtrackProFX.beatJumpValues = [
+    "0.0625_",
+    "0.125_",
+    "0.25_",
+    "0.5_",
+    "1_",
+    "2_",
+    "", // "beatjump_forward"/"beatjump_backward" - jump by the value selected in Mixxx GUI (4 by default)
+    "8_"
+];
+
+// pad modes, don't touch
+MixtrackProFX.PadModeControls = {
+    HOTCUE: 0x00,
+    AUTOLOOP: 0x0D,
+    FADERCUTS: 0x07,
+    SAMPLE1: 0x0B,
+    BEATJUMP: 0x02,
+    SAMPLE2: 0x0F
+};
+
+// state variables, don't touch
+MixtrackProFX.shifted = false;
+
+MixtrackProFX.init = function() {
+    // initialize component containers
+    MixtrackProFX.deck = new components.ComponentContainer();
+    MixtrackProFX.effect = new components.ComponentContainer();
+    var i;
+    for (i = 0; i < 2; i++) {
+        MixtrackProFX.deck[i] = new MixtrackProFX.Deck(i + 1);
+        MixtrackProFX.effect[i] = new MixtrackProFX.EffectUnit(i + 1);
+    }
+
+    MixtrackProFX.browse = new MixtrackProFX.Browse();
+    MixtrackProFX.gains = new MixtrackProFX.Gains();
+
+    // send sysexes
+    var exitDemoSysex = [0xF0, 0x7E, 0x00, 0x06, 0x01, 0xF7];
+    midi.sendSysexMsg(exitDemoSysex, exitDemoSysex.length);
+
+    var statusSysex = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7];
+    midi.sendSysexMsg(statusSysex, statusSysex.length);
+
+    // enables 4 bottom pads "fader cuts"
+    var faderCutSysex = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0xF7];
+    midi.sendSysexMsg(faderCutSysex, faderCutSysex.length);
+
+    // initialize leds
+    for (i = 0; i < 2; i++) {
+        midi.sendShortMsg(0x94 + i, 0x00, 0x7F); // hotcue
+        midi.sendShortMsg(0x94 + i, 0x0D, 0x01); // auto loop
+        midi.sendShortMsg(0x94 + i, 0x07, 0x01); // "fader cuts"
+        midi.sendShortMsg(0x94 + i, 0x0B, 0x01); // sample1
+
+        // shifted leds
+        midi.sendShortMsg(0x94 + i, 0x0F, 0x01); // sample2
+        midi.sendShortMsg(0x94 + i, 0x02, 0x01); // beatjump
+    }
```

Comment:
move this to the padsection constructor instead?

Reply:
When I moved it to PadSection constructor, the lights are not initialized. I am certain I'm sending the same MIDI codes, I see them in the debug output. Yet the lights are all 0x00. Any ideas why this might be happening?

Reply:
Maybe the sysexes interfere with that? What if you send the exitDemoSysex before instancing the deck. so send it at the very start of the init function?

Reply:
That was it, thanks!